internal/topology/variables/cluster_variable_defaulting_test.go

```diff
@@ -29,15 +29,15 @@ import (
 func Test_DefaultClusterVariables(t *testing.T) {
 	tests := []struct {
```

Comment:
Just something to think about:

Do we have test coverage (either in defaulting or validation) for the cases where we "transition" between the various ways variables can be / have to be defined:
* 1 global variable
* namespaced variables because of conflict
* namespaced variables because users chose too

With transition I mean that the Cluster.spec.topology variables is using one of those options and based on variable definitions we default/validate based on another one. (for all permutations)

Could be that we already have all permutations and a lot of those are probably just error cases. Would be great if you can go through the permutations and double check that we have them all covered.

Reply:
It would be great if we can order / group the test cases in a way that's easier to assess if we covered all cases (+ maybe add a quick godoc comment over each section) (starting here as everything above are the straightforward cases)

E.g. One group for each of
* conflicting definitions
* non-conflicting definitions
(not sure if that's everything)

And then go through each group for the various variants how the Cluster.spec.topology.variables could look like:
* no variable defined at all
* a global one defined
* a namespaced one defined
* all namespaced ones defined (no-op)
* ...


Would be nice if we can also write the names in a consistent way. Currently it's not easy to figure out which cases we cover and if there are gaps

(same for validation tests)

Reply:
One additional point. Do we have enough coverage for cases where for one variable name some definitions are optional and others are required? (for defaulting and validation)

Reply:
>With transition I mean that the Cluster.spec.topology variables is using one of those options and based on variable definitions we default/validate based on another one. (for all permutations)

Not sure I understand what the ask is here. I'm currently cleaning up the tests to be more ordered. Let's sync on whether they're good enough once I've pushed that change (+/- time to review)